### PR TITLE
updated code for theme and navbar theme icon.

### PR DESCRIPTION
**Pull request by Ritika8081**

### CODE REVIEW

File: src/components/Canvas.tsx [hunk 1]
```diff
@@ -14,7 +15,7 @@ interface CanvasProps {
   data: string;
   selectedBits: BitSelection;
   isGridView: boolean;
-  isDisplay: boolean;
+  isDisplay: boolean; // New prop for play/pause functionality
```
Comment (akadeepesh) [inline src/components/Canvas.tsx]: Edit this, or add comment to all props

File: src/components/Canvas.tsx [hunk 2]
```diff
@@ -28,6 +29,7 @@ const Canvas: React.FC<CanvasProps> = ({
   const chartRef = useRef<SmoothieChart[]>([]);
   const seriesRef = useRef<(TimeSeries | null)[]>([]);
   const [isChartInitialized, setIsChartInitialized] = useState(false);
+  const [isPaused, setIsPaused] = useState(Array(channels.length).fill(false)); // Paused state for each channe
```
Comment (akadeepesh) [inline src/components/Canvas.tsx]: Remove this we don't need this

File: src/components/Canvas.tsx [hunk 3]
```diff
@@ -138,7 +134,7 @@ const Canvas: React.FC<CanvasProps> = ({
 
     batchBuffer.forEach((batch) => {
       channels.forEach((channel, index) => {
-        if (channel) {
+        if (channel && !isPaused[index]) {
```
Comment (akadeepesh) [inline src/components/Canvas.tsx]: Remove isPaused

File: src/components/Canvas.tsx [hunk 4]
```diff
@@ -148,7 +144,7 @@ const Canvas: React.FC<CanvasProps> = ({
     });
 
     batchBuffer.length = 0;
-  }, [channels, batchBuffer, isGlobalPaused]);
+  }, [channels, batchBuffer, isGlobalPaused, isPaused]);
```
Comment (akadeepesh) [inline src/components/Canvas.tsx]: Remove isPaused

File: src/components/Canvas.tsx [hunk 5]
```diff
@@ -165,14 +161,21 @@ const Canvas: React.FC<CanvasProps> = ({
     },
     [processBatch, batchBuffer, isDisplay]
   );
-
+  // console.log(data);
```
Comment (akadeepesh) [inline src/components/Canvas.tsx]: remove

File: src/components/Canvas.tsx [hunk 6]
```diff
@@ -289,15 +313,19 @@ const Canvas: React.FC<CanvasProps> = ({
     <div className="flex flex-col justify-center items-start px-4 m-4 h-[80vh]">
       <div
         className={`grid ${
-          isGridView ? "md:grid-cols-2 grid-cols-1" : "grid-cols-1"
+          isGridView
+            ? "md:grid-cols-2 grid-cols-1" // Apply the same spacing for both horizontal and vertical gaps
+            : "grid-cols-1"
```
Comment (akadeepesh) [inline src/components/Canvas.tsx]: Remove comments from here

File: src/components/Canvas.tsx [hunk 7]
```diff
@@ -289,15 +313,19 @@ const Canvas: React.FC<CanvasProps> = ({
...
       >
         {channels.map((channel, index) => {
           if (channel) {
             return (
               <div
                 key={index}
-                className={`flex flex-col w-full relative h-full`}
+                className={`flex flex-col w-full relative h-full${
+                  isGridView ? "" : ""
+                }`}
```
Comment (akadeepesh) [inline src/components/Canvas.tsx]: :)

File: src/components/Connection.tsx [hunk 8]
```diff
@@ -163,7 +163,7 @@ const Connection: React.FC<ConnectionProps> = ({
     // Function to connect to the device
     try {
       const port = await navigator.serial.requestPort(); // Request the serial port
-      await port.open({ baudRate: 57600 }); // Open the port with baud rate 57600
+      await port.open({ baudRate: 115200 }); // Open the port with baud rate 57600
```
Comment (akadeepesh) [inline src/components/Connection.tsx:166]: baudRate fixing

File: src/components/LandingComp/HeadSection.tsx [hunk 9]
```diff
@@ -7,13 +8,30 @@ import Image from "next/image";
 import { useTheme } from "next-themes";
 import Chords from "./Chords";
 
-const HeadSection = () => {
-  const { theme } = useTheme();
+const HeadSection: React.FC = () => {
+  const { resolvedTheme } = useTheme(); // `resolvedTheme` reflects the system theme or the user-set theme
```
Comment (akadeepesh) [inline src/components/LandingComp/HeadSection.tsx]: Remove comment

File: src/components/Theming/mode-toggle.tsx [hunk 10]
```diff
@@ -1,39 +1,25 @@
-"use client";
-
-import React, { useEffect } from "react";
-import { MoonIcon, SunIcon } from "@radix-ui/react-icons";
 import { useTheme } from "next-themes";
+import { MoonIcon, SunIcon } from "@radix-ui/react-icons";
+import { Button } from "@/components/ui/button"; // Importing the Button from shadcn
```
Comment (akadeepesh) [inline src/components/Theming/mode-toggle.tsx]: remove this comment

File: src/components/Theming/mode-toggle.tsx [hunk 11]
```diff
@@ -1,39 +1,25 @@
...
-        )}
-      </Button>
-    </div>
+    <Button variant="ghost" size="sm" onClick={toggleTheme}>
+      {currentTheme === "dark" ? (
+        <SunIcon width={16} height={16} />
+      ) : (
+        <MoonIcon width={16} height={16} />
+      )}
+    </Button>
```
Comment (akadeepesh) [inline src/components/Theming/mode-toggle.tsx]: keep the outer div wrap, it will give us access to add external style and props

**Review by akadeepesh — CHANGES_REQUESTED** [review]

**Comment by Ritika8081** [comment]
I have fixed issues, please check.

**Pull request merged**